packages/mobx-state-tree/__tests__/core/model.test.ts

```diff
@@ -120,25 +120,43 @@ describe("Edge case behavior", () => {
             const modelSnapshot = getSnapshot(Model.create())
             expect(modelSnapshot).toEqual({})
         })
+        if (process.env.NODE_ENV !== "production") {
```

Comment:
**suggestion:** @nithinssabu - I added a test here to ensure we aren't throwing new errors to people who might be doing some edge-case behavior like instantiating empty models entirely.

Will you consider taking a look for other scenarios, like when a user instantiates a model *only* with a valid object, but no string, and make sure we aren't throwing new errors when we shouldn't?

My intention is mostly to make sure we have some test coverage around: `if (devMode() && typeof args[0] !== "string" && args[1])` which ensures the condition operates as you expect it/will let us know if anyone accidentally modifies it in a way that will break.

If you don't have time or interest to make this change, this is not a blocker for me. I'll add some additional tests later as well.

Reply:
@coolsoftwaretyler Since the behaviour that it should not throw an error when we provide no arguments to the function is valid for both dev and production, do we need the condition `if (process.env.NODE_ENV !== "production")` for this test case?

Thinking about it further, isn't it implicity tested by other cases under the same `describe` block? When we test that the `model will be named AnonymoustModel` and `the model will have no properties`, we are testing that a model is successfully created and it does not throw an error, right?

Reply:
@nithinssabu - you are correct, but one thing I'm hoping to improve in MST is to have a higher quantity of tests at a more granular level of specificity. I want the test suite to function as both a set of automated tests, and a way for us to communicate intent. 

So in this case, I want to specifically communicate externally that in development mode, calling `types.model()` does not throw an error.